src/ucp/core/ucp_ep.c

```diff
@@ -139,36 +149,67 @@ ucs_status_t ucp_ep_create_base(ucp_worker_h worker, const char *peer_name,
     ucs_debug("created ep %p to %s %s", ep, ucp_ep_peer_name(ep), message);
     return UCS_OK;
 
+err_free_keys:
+    ucs_free(ucp_ep_ext_gen(ep)->ids);
 err_free_ep:
     ucs_strided_alloc_put(&worker->ep_alloc, ep);
 err:
     return status;
 }
 
-ucs_status_t ucp_worker_create_ep(ucp_worker_h worker, const char *peer_name,
-                                  const char *message, ucp_ep_h *ep_p)
+static void ucp_ep_destroy_base(ucp_ep_h ep)
+{
+    ucs_free(ucp_ep_ext_gen(ep)->ids);
+    ucs_strided_alloc_put(&ep->worker->ep_alloc, ep);
+}
+
+ucs_status_t ucp_worker_create_ep(ucp_worker_h worker, unsigned ep_init_flags,
+                                  const char *peer_name, const char *message,
+                                  ucp_ep_h *ep_p)
 {
     ucs_status_t status;
     ucp_ep_h ep;
 
     status = ucp_ep_create_base(worker, peer_name, message, &ep);
     if (status != UCS_OK) {
-        return status;
+        goto err;
     }
 
     ucs_list_add_tail(&worker->all_eps, &ucp_ep_ext_gen(ep)->ep_list);
 
+    ucs_assert(ucp_ep_ext_gen(ep)->ids->local == UCP_EP_ID_INVALID);
+    status = ucs_ptr_map_put(&worker->ptr_map, ep,
+                             !!(ep_init_flags &
```

Comment:
no need for `!!`
` * @param [in]  indirect  If nonzero, the pointer @a ptr is stored in the` - **nonzero**

Reply:
it can work incorrectly without `!!` if flags larger sizeof(int)

Reply:
we have such code in may places ... and i think coverity/cppcheck will warn in such case

Reply:
not sure if coverity/cppcheck warns on such errors but I had a fun to debug thinks like this in UCX a few times

Reply:
ok